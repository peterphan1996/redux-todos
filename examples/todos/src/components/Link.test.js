import React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';

const setup = setupProps => {
  const defaultProps = {
    active: false,
    onClick: jest.fn(),
    children: 'test',
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <Link active={props.active} onClick={props.onClick}>
      {props.children}
    </Link>
  );

  return {
    props,
    wrapper,
  };
};

describe('Link', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('renders a disabled button when active is true', () => {
    const { wrapper } = setup({ active: true });
    expect(wrapper.find('button').prop('disabled')).toEqual(true);

  });

  describe('when button is clicked', () => {
    test('should trigger click event correctly', () => {
      const { props, wrapper } = setup();
      expect(wrapper).toMatchSnapshot();

      const link = wrapper.find('button');
      link.simulate('click');

      expect(props.onClick).toHaveBeenCalledTimes(1);
    });
  });
});