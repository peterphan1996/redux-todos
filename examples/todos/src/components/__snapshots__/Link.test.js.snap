// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders without crashing 1`] = `
<button
  disabled={false}
  onClick={[MockFunction]}
  style={
    Object {
      "marginLeft": "4px",
    }
  }
>
  test
</button>
`;

exports[`Link when button is clicked should trigger click event correctly 1`] = `
<button
  disabled={false}
  onClick={[MockFunction]}
  style={
    Object {
      "marginLeft": "4px",
    }
  }
>
  test
</button>
`;
